### PR TITLE
Visualiser full lyceum

### PR DESCRIPTION
**Pull request by nic-barbara**
Added in the visualiser from [`LyceumMuJoCoViz.jl`](https://github.com/Lyceum/LyceumMuJoCoViz.jl) as a package extension to our `MuJoCo.jl`. I've tested this on mac and ubuntu - yet to be test on windows at the time of writing.

We currently have feature parity with the old Lyceum visualiser apart from a few key functions:
- They have a function to `reset!()` a MuJoCo environment when the user presses `backspace`. I have excluded this, since resetting is very specific to each model/experiment and doesn't really make sense for a generic MuJoCo model. It made sense in the old visualiser, which used `gym`-like MuJoCo environments.
- I have removed the `Trajectory` mode allowing the user to replay state trajectories through the visualiser. We may want to add this back in later. It relies on the `state` of a MuJoCo model being defined. Again, this makes sense for `gym`-like environments, but not for generic MuJoCo `Model` and `Data` objects.

A few C globals were also required to add button and mouse callbacks. These are currently defined in `ext/VisualiserExt.jl`, but maybe it's better to define them elsewhere. Suggestions would be appreciated :)

### CODE REVIEW

**Comment by nic-barbara** [comment]
This pull request is currently a draft. If I have time in the next day or two, I'll add in the option to apply forces to a model in the visualiser by clicking on it. See [here](https://github.com/rohanpsingh/mujoco-python-viewer) for an example.

**Comment by nic-barbara** [comment]
There seems to be an issue with `MuJoCo_jll` in the CI, any ideas @JamieMair?

**Comment by JamieMair** [comment]
> There seems to be an issue with `MuJoCo_jll` in the CI, any ideas @JamieMair?

It's because the MuJoCo_jll package was updated and has a different UUID, I got the same error on my machine and had to remove some cached files in my `.julia` folder. Once the Yggdrasil PR goes in, we can avoid manually pointing to the jll package as it should be registered, which will let us fix the CI error.

**Comment by nic-barbara** [comment]
> This pull request is currently a draft. If I have time in the next day or two, I'll add in the option to apply forces to a model in the visualiser by clicking on it. See [here](https://github.com/rohanpsingh/mujoco-python-viewer) for an example.

Turns out this is actually already implemented in the original `LyceumMuJoCoViz.jl`, it's just not documented anywhere. Works great. I'll add some documentation on the mouse and button callbacks.

File: ext/VisualiserExt/VisualiserExt.jl [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+struct CRef{name, T}
+    CRef{name, T}() where {name, T} = new{name::Symbol, T}()
+end
+
+Base.getindex(::CRef{name,T}) where {name,T} = loadglobal(Val(name), T)
+
+function loadglobal(::Val{name}, ::Type{T}, i::Integer = 1) where {name,T}
+    Base.unsafe_load(getglobal(Val(name), T))
+end
+
```
Comment (JamieMair) [inline ext/VisualiserExt/VisualiserExt.jl:70]: It took me a while to understand what this is doing, but yes, this is completely fine. I've moved this part to another file for clarity.

File: ext/VisualiserExt/modes.jl [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+    println(io1, "Realtime Factor")
+    @printf io2 "%.2fx\n" x.realtimefactor
+    return nothing
+end
+
+# TODO: Implement Trajectory mode later. It relies on setstate!(m, d), which is highly dependent on the model and should be user-defined. We should think about the best way to include this.
+
+####
+#### Trajectory
+####
```
Comment (JamieMair) [inline ext/VisualiserExt/modes.jl:94]: I have created https://github.com/JamieMair/MuJoCo.jl/issues/41 to keep track of this, and removed the commented out code for now in my branch of this pr

File: ext/VisualiserExt/VisualiserExt.jl [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+function MuJoCo.Visualiser.test_visualiser()
+    root_path = normpath(joinpath(@__DIR__, "..", ".."))
+
+    # Load model, data
+    model = load_xml(joinpath(root_path, "models", "humanoid.xml"))
+    data  = init_data(model)
+
+    # # Change initial conditions for cartpole
+    # data.qpos .= @SVector [0, 0.0001]
+    # data.qvel .= @SVector [0.01, 0]
```
Comment (JamieMair) [inline ext/VisualiserExt/VisualiserExt.jl:128]: Removed commented code in my branch

**Review by JamieMair — CHANGES_REQUESTED** [review]
This is great, we can probably merge this now. I think the trajectory stuff can be postponed until later. I have created some issues for doing just that. I have just removed some of the commented out code in favour of GH issues for todos - see https://github.com/JamieMair/MuJoCo.jl/pull/44

**Pull request merged**